### ISSUE TITLE
Add pie to replace pecl/pear

### ISSUE THREAD

**Issue by jnoordsij** [docker-library/php#1554]
Recently the PHP foundation has been working on a new official tool for installing extensions, which has now been released as PHP PIE. See also https://thephp.foundation/blog/2024/11/19/pie-pre-release/ and https://github.com/php/pie. Given that this aims to replace PECL (and indirectly PEAR), which currently are already deprecated and are causing more and more issues (see e.g. #1552 and #1553), the official image could probably follow the same path at some point.

Given that the tool is currently still in pre-release state and still requires more testing and moreover requires the PHP extension ecosystem to [make some active changes](https://github.com/php/pie/blob/main/docs/extension-maintainers.md) to allow installation, the time to migrate is most likely not now just yet. However, I am hoping to slowly start discussing the possibilities and required steps with this issue.

**Comment by tianon** [comment]
Nice, this is good to see!

However, I think we'll definitely wait until it's more well-baked / official within the community before considering it, but I'm honestly also kind of inclined to not include it unless/until it's part of the standard PHP install (which is the only reason we include PECL / PEAR today).

**Comment by jnoordsij** [comment]
Leaving it out also sounds quite reasonable; probably the first course of action then, when things stabilize a bit more, is adjusting the docs to list PIE as an alternative/preferred way to install extensions and make note of the deprecation of PECL/PEAR installs.

Wondering though, it seems that PECL/PEAR could also just be installed on their own without including them in the base image. So it feels that there's still some choice here? Although if there's no option to bundle PIE at installation time in a future version, probably not doing that makes more sense. And then keeping PECL/PEAR around until downstream finally removes it is the least disruptive course of action.

**Comment by tianon** [comment]
Yeah, we've historically included PECL/PEAR because a "default install" includes them, and we've been explicit about including that configure flag to try and help manage the transition to deprecation, but once it's gone, we definitely don't plan to sideload them or anything like that (especially as related to our experience with similar discussions in Ruby, Python, and Node.js communities who've also had arguments against the "default included" package managers).

Including a blurb in the documentation is totally sane though. :+1:

**Comment by dmitryuk** [comment]
Still relevant
